cfme/services/myservice/__init__.py

```diff
@@ -34,6 +35,19 @@ def __init__(self, appliance, name=None, description=None, vm_name=None):
         self.vm_name = vm_name
         self.parent = self.appliance.context
 
+    @property
+    def rest_obj(self):
+        return self.appliance.rest_api.collections.services.find_by(name=self.name)[0]
+
+    def add_resource_generic_object(self, gen_obj):
```

Comment:
Since there are already contextual methods defined for `MyService`, I think this should be added to a new `rest.py` context module.  

To do this you'll need to do a few things:
1. Add a contextualmethod above with the others, named as you wish
2. Create a new `rest.py` module with `ui.py` and `ssui.py`
3. Import this new module below with importscan
4. Decorate and define the method in `rest.py`

When the method is called, it'll resolve and call the ViaREST context automatically - or if you define this method for all three contexts, a test could be parametrized to test all three interfaces with the same test function.